scripts/test/AbsorptionShapesTest.py

```diff
@@ -0,0 +1,222 @@
+        ash.material = {'ScatteringXSection':20}
+        res = ash.material;
+        self.assertEqual(res['AttenuationXSection'],0.5)
+        self.assertEqual(res['ScatteringXSection'],20)
+        self.assertEqual(len(res),4)
+
+    def test_adsrp_cylinder(self):
+        ash = Cylinder('V',[10,2])
+        res = ash.shape
+        self.assertEqual(res['Height'],10)
+        self.assertEqual(res['Radius'],2)
+
+        ash.shape = [5,1,[0,1,0],[0.,0.,-0.5]]
+        res = ash.shape;
+        self.assertEqual(res['Height'],5)
+        self.assertEqual(res['Radius'],1)
+        self.assertEqual(res['Axis'],[0,1,0])
+        self.assertEqual(res['Center'],[0,0,-0.5])
+
+        ash.shape = {'Height':5,'Radius':2,'Axis':[1,0,0],'Center':[0.,0.,0.]}
+        res = ash.shape;
+        self.assertEqual(res['Height'],5)
+        self.assertEqual(res['Radius'],2)
+        self.assertEqual(res['Axis'],[1,0,0])
+        self.assertEqual(res['Center'],[0,0,0])
+
+        test_ws = CreateSampleWorkspace(NumBanks=1,BankPixelWidth=1)
+        test_ws = ConvertUnits(test_ws,'DeltaE',Emode='Direct',EFixed=2000)
+        #cor_ws,corrections = ash.correct_absorption(test_ws)
+        #n_bins = corrections.blocksize()
+        #corr_ranges = [n_bins,corrections.readY(0)[0],corrections.readY(0)[n_bins-1]]
+        #np.testing.assert_almost_equal(corr_ranges,[97,0.2258,0],4)
+        mccor_ws,mc_corr = ash.correct_absorption(test_ws,NumberOfWavelengthPoints=20)
+        n_bins = mc_corr.blocksize()
+        mccorr_ranges = [n_bins,mc_corr.readY(0)[0],mc_corr.readY(0)[n_bins-1]]
+        np.testing.assert_almost_equal(mccorr_ranges ,[97,0.2657,0.0271],4)
+
+    def test_adsrp_Plate(self):
+        ash = FlatPlate('V',[10,2,0.1])
+        res = ash.shape
+        self.assertEqual(res['Height'],10)
+        self.assertEqual(res['Width'],2)
+        self.assertEqual(res['Thick'],0.1)
+
+        ash.shape = [5,1,0.2,[0,1,0],10]
+        res = ash.shape;
+        self.assertEqual(res['Height'],5)
+        self.assertEqual(res['Width'],1)
+        self.assertEqual(res['Thick'],0.2)
+        self.assertEqual(res['Center'],[0,1,0])
+        self.assertEqual(res['Angle'],10)
+
+        ash.shape = {'Height':5,'Width':1,'Thick':2,'Center':[0.,0.,0.],'Angle':20}
+        res = ash.shape;
+        self.assertEqual(res['Height'],5)
+        self.assertEqual(res['Width'],1)
+        self.assertEqual(res['Thick'],2)
+        self.assertEqual(res['Center'],[0,0,0])
+        self.assertEqual(res['Angle'],20)
+
+        test_ws = CreateSampleWorkspace(NumBanks=1,BankPixelWidth=1)
+        test_ws = ConvertUnits(test_ws,'DeltaE',Emode='Direct',EFixed=2000)
+        #cor_ws,corrections = ash.correct_absorption(test_ws,ElementSize=5)
+        #n_bins = corrections.blocksize()
+        #corr_ranges = [n_bins,corrections.readY(0)[0],corrections.readY(0)[n_bins-1]]
+        #np.testing.assert_almost_equal(corr_ranges,[97,0.4504,0.0009],4)
```

Comment:
Should these commented out lines be removed? I guess they're using the "fast" algorithm? Why not uncomment them and let them run?